### ISSUE TITLE
How does the encoder work exactly?

### ISSUE THREAD

**Issue by r-luo** [UTCAA-Mentorship-2018/tianyudu#2]
I haven't used this encoder before, how does it work?
https://github.com/UTCAA-Mentorship-2018/tianyudu/blob/master/data_proc.py#L146-L150

**Comment by TianyuDu** [comment]
It's just a normal integer encoder. I found the data types from the csv file are int64, float64 and object(actually string). And the encoder maps all string into integers.

**Comment by r-luo** [comment]
I see. How would your model deal with the following case?
```
Suppose your data have 4 classes, and each of them are associated with the target in the following way:
  A: strongly positive
  B: strongly negative
  C: strongly positive
  D: strongly negative
and the encoding is A=1, B=2, C=3, D=4
```

**Comment by TianyuDu** [comment]
Did you mean strongly/weakly positive/negative?
In this case, since there are obvious ordinal correlations among categories, I would encode it as
"Strongly positive" encoded as 2
"Weakly positive" encoded as 1
"Weakly negative" encoded as -1
"Strongly negative" encoded as -2

In other cases where no obvious ordinal correlation presents, like loan type, I would simply encode categories into integers.

**Comment by r-luo** [comment]
Right, so what happens to your model when you don't know the ordinal correlations (e.g. loan type as you mentioned)? I'll ask you to explain from 3 different stand points - linear models (e.g. linear/logistic regression), GBM and Neural Networks.

**Comment by TianyuDu** [comment]
For now, the label encoder just loops over all observations and label the first category met as 0 and the second distinct one as 1 and so on.

**Comment by r-luo** [comment]
But can you answer the question of how each type of models that I mentioned would deal with the way your current encoder works?
